src/vsg/io/FileSystem.cpp

```diff
@@ -17,7 +17,7 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 
 #include <cstdio>
 
-#if defined(WIN32) && !defined(__CYGWIN__)
+#if (defined(WIN32) || defined(_WIN32)) && !defined(__CYGWIN__)
```

Comment:
I think we probably just want to check `_WIN32` (the thing the compiler's set to tell us that the OS is Windows that all compilers set on Windows) rather than also checking `WIN32` (which usually comes from `Windows.h` and might not be set reliably). The VSG seems to use both without a clear distinction why each one was picked in each situation.

There are also a bunch of other checks elsewhere in the project, which may also need tweaking. You might need to take care with Cygwin, though, as it's pretty good at pretending to be Unix, so some Windows details might be hidden and it might be preferable to use the non-Windows code.